build.gradle

```diff
@@ -1,23 +1,102 @@
 import org.apache.tools.ant.taskdefs.condition.Os
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.Properties;
 
 plugins {
     id "java"
     // moditect for module-info.class injection and jlink
     id "org.moditect.gradleplugin" version "1.0.0-rc3"
     // jpackage plugin for packging jextract native app
     id "org.panteleyev.jpackageplugin" version "1.3.1"
+    // for downloading files
+    id "de.undercouch.download" version "5.0.2"
 }
 
 version = '1.0'
+project.ext.minimalJDKVersion = Runtime.Version.parse("18");
+project.ext.depsDir = "deps"
 
 def static checkPath(String p) {
-    if (!Files.exists(Path.of(p))) {
+    checkPath(Path.of(p));
+}
+
+def static checkPath(Path p) {
+    if (!Files.exists(p)) {
         throw new IllegalArgumentException("Error: the path ${p} does not exist");
     }
 }
 
+def static Runtime.Version getJDKVersion(String jdkHome) {
+    Path releaseFile = Path.of(jdkHome).resolve("release");
+    checkPath(releaseFile);
+
+    Properties props = new Properties();
+    try (InputStream input = Files.newInputStream(releaseFile)) {
+        props.load(input);
+    }
+
+    String foundVersionStr = props.getProperty("JAVA_VERSION");
+    if (foundVersionStr == null) {
+        throw new IllegalArgumentException("Could not determine java version from ${releaseFile}");
+    }
+
+    foundVersionStr = foundVersionStr.substring(1, foundVersionStr.length() - 1); // version in release file is quoted
+    Runtime.Version foundVersion = Runtime.Version.parse(foundVersionStr);
+
+    return foundVersion;
+}
+
+def static checkJDKRelease(String jdkHome, Runtime.Version expectedVersion) {
+    Runtime.Version foundVersion = getJDKVersion(jdkHome);
+    if (foundVersion.compareToIgnoreOptional(expectedVersion) < 0) {
+        throw new IllegalArgumentException("Incorrect build JDK version. " +
+            "Expected at least '${expectedVersion}', but found '${foundVersion}'");
+    }
+}
+
+def String downloadJDK() {
+    String jdkUrl = "https://download.java.net/java/GA/jdk18/43f95e8614114aeaa8e8a5fcf20a682d/36/GPL/openjdk-18_";
```

Comment:
You should put this in a property so gradle won't have to recompile/reconfigure the script if this value changes.